app/views/admins/_user_entry.haml

```diff
@@ -79,3 +97,33 @@
               method: :post,
               data: {confirm: t("admins.user_search.are_you_sure_lock_account")},
               class: "btn btn-danger btn-block"
+          - if Role.is_admin?(user.person)
+            = link_to t("admins.user_search.remove_admin"),
+              admin_remove_admin_path(user),
+              method: :post,
+              class: "btn btn-danger btn-block"
+          - else
+            = link_to t("admins.user_search.make_admin"),
+              admin_make_admin_path(user),
+              method: :post,
+              class: "btn btn-info btn-block"
+          - if Role.moderator?(user.person)
```

Comment:
This has an edge case. The `moderator?` function checks if the list of moderators includes the current user. Unfortunately, the list of moderators is generated with

https://github.com/diaspora/diaspora/blob/883eb54ecfb9b67e12b7072304473ce0d637f65c/app/models/role.rb#L12

so it includes admins as well. Given a user that only has an admin role, the UI would still show 
<img width="114" alt="screen shot 2018-09-30 at 18 27 02" src="https://user-images.githubusercontent.com/344777/46259859-aa13ac80-c4de-11e8-82e6-608ca7cd028a.png"> and allow the "Remove moderator" button to be clicked. On clicking, the application explodes, since there is no role to remove. So this needs a different check - or the check to be altered without breaking other behavior.